src/workerd/api/r2-test.js

```diff
@@ -0,0 +1,89 @@
+// Copyright (c) 2023 Cloudflare, Inc.
+// Licensed under the Apache 2.0 license found in the LICENSE file or at:
+//     https://opensource.org/licenses/Apache-2.0
+
+import assert from 'node:assert';
+
+export default {
+  // Handler for HTTP request binding makes to R2
+  async fetch(request, env, ctx) {
+    // We only expect PUT/Get
+    assert(request.method === "PUT" || request.method === "GET")
+
+    // Each request should have a metadata size header indicating how much
+    // we should read to understand what type of request this is
+    const metadataSizeString = request.headers.get("cf-r2-metadata-size")
+    assert(metadataSizeString !== null)
+    const metadataSize = parseInt(metadataSizeString)
+    assert(!Number.isNaN(metadataSize))
+
+    const reader = request.body.getReader({ mode: 'byob' })
+    const jsonArray = new Uint8Array(metadataSize)
+    const { value } = await reader.readAtLeast(metadataSize, jsonArray)
+    const jsonRequest = JSON.parse(new TextDecoder().decode(value))
+
+    // Currently not using the body in these test so I'm going to just discard
+    while (true) {
+      const read = await reader.read(new Uint8Array(65536))
+      if (!read.done) {
+        continue
+      } else {
+        break
+      }
+    }
+
+    // Assert it's the correct version
+    assert(jsonRequest.version = 1)
+
+    if (request.method === "PUT") {
+      assert(jsonRequest.method === "put")
+
+      if (jsonRequest.object === "onlyIfStrongEtag") {
+        assert(jsonRequest.onlyIf.etagMatches.length === 1)
```

Comment:
Since you're using `node:assert`, it is better to use `assert.strictEqual(...)` here instead of `assert(...)`

Reply:
You could also probably greatly simplify things using `assert.deepStrictEqual(...)` but that's only a guess.

Reply:
That's a lot cleaner, thanks for the tip!